packages/editor/src/browser/editor-command.ts

```diff
@@ -163,6 +163,17 @@ export class EditorCommandContribution implements CommandContribution {
     @inject(ResourceProvider)
     protected readonly resourceProvider: ResourceProvider;
 
+    @postConstruct()
+    protected async init(): Promise<void> {
+        this.editorPreferences.onPreferenceChanged(e => {
+            if (e.preferenceName === 'editor.autoSave' && e.newValue === 'on') {
+                this.editorManager.all
```

Comment:
why do we save only editors? we should save all saveable. It won't work for custom editors and any widgets using resources.

Reply:
> why do we save only editors? we should save all saveable. It won't work for custom editors and any widgets using resources.

@akosyakov I updated it to use the original approach of using the `shell` to `saveAll`.